P1/Project_Code/frontend/src/components/Media_Cards/MediaCard.jsx

```diff
@@ -1,59 +1,50 @@
-import { useState, useEffect } from "react";
+// src/components/MediaCard.jsx
+import { useState, useEffect } from 'react';
+import { getImagePath } from '../../services/getImagePath.js';
 
 const MediaCard = ({ movie, type }) => {
+    const [imgSrc, setImgSrc] = useState('');
 
-    const [imgSrc, setImgSrc] = useState("");
-
-    // Dynamically import the image required for the media card
+    // Get the static path to the image required for the media card
     useEffect(() => {
-        const loadPic = async () => {
-            const pic = await import(`../../assets/thumbnails/${movie.pics.regular.medium}`);
-            setImgSrc(pic.default);
-        }
-
-        loadPic();
-    }, []);
+        const imagePath = getImagePath(movie.pics.regular.medium);
+        setImgSrc(imagePath);
+    }, [movie]);
 
     return (
         <>
             {
-                imgSrc
-                ?(
-                    type.toLowerCase() == "trending"
-                    ?
-                        (   <div style={{ borderRadius: "20px", width: "540px", height: "280px", backgroundImage: `url(${imgSrc})`, backgroundPosition: "center", backgroundRepeat: "no-repeat", backgroundSize: "cover"}}>
-                                <div className="trending-media-info">
-                                    <div className="media-info">
-                                        <h4 className="media-info-item">{movie.year} •</h4>
-                                        <h4 className="media-info-item">{movie.category} •</h4>
-                                        <h4 className="media-info-item">{movie.rating}</h4>
-                                    </div>
-                                    <div className="media-title">{movie.title}</div>
-                                </div>
-                            </div>
-                        )
-                    :
-
-                        (
-                            <div>
-                                <img src={imgSrc} alt={movie.title} />
+                imgSrc ? (
```

Comment:
This large ternary operator is hard to read. I suggest refactoring this code and decomposing the logic.